src/Middleware/RequestThrottling/test/MiddlewareTests.cs

```diff
@@ -73,36 +74,62 @@ public async void RequestsBlockedIfQueueFull()
         [Fact]
         public void IncomingRequestsFillUpQueue()
         {
-            var middleware = TestUtils.CreateTestMiddleware(
-                queue: TestStrategy.AlwaysBlock);
+            var testQueue = TestPolicy.AlwaysBlock;
+            var middleware = TestUtils.CreateTestMiddleware(testQueue);
 
-            Assert.Equal(0, middleware.QueuedRequestCount);
+            Assert.Equal(0, testQueue.QueuedRequests);
 
             _ = middleware.Invoke(new DefaultHttpContext());
-            Assert.Equal(1, middleware.QueuedRequestCount);
+            Assert.Equal(1, testQueue.QueuedRequests);
 
             _ = middleware.Invoke(new DefaultHttpContext());
-            Assert.Equal(2, middleware.QueuedRequestCount);
+            Assert.Equal(2, testQueue.QueuedRequests);
+        }
+
+        [Fact]
+        public void EventCountersTrackQueuedRequests()
+        {
+            var blocker = new TaskCompletionSource<bool>();
+
+            // requires
+            var testQueue = new TestPolicy(
```

Comment:
I feel like these `testQueue` constructors aren't clear about what is being tested. Maybe it's the name of TestPolicy or something.  Also, maybe changing invoke to onInvoke?